include/device.h

```diff
@@ -450,6 +451,29 @@ static inline int device_get_power_state(struct device *device,
 						 NULL, NULL);
 }
 
+/**
+ * @brief Call the get scheduled power state function of a device
+ *
+ * This function lets the caller know the current device scheduled
+ * suspend state based on the upcoming kernel idle ticks. This state
+ * will be one of the defined suspend states allowed for the devices
+ * in that system.
+ *
+ * @param device Pointer to device structure of the driver instance.
+ * @param device_power_state Device power state to be filled by the device.
+ * @param ticks The upcoming kernel idle time.
+ *
+ * @retval 0 If successful.
+ * @retval Errno Negative errno code if failure.
+ */
+static inline int device_get_scheduled_power_state(struct device *device,
```

Comment:
I am not sure that power state selection should be done by device drivers. The driver does not have knowledge about system state so it will be hard to implement reasonable algorithm there.

Reply:
can we just pass the state system going to or suspend ticks to devices? @pizi-nordic